packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -1570,10 +1571,45 @@ export const $markAsPageable: MarkAsPageableDecorator = (
       target: context.decoratorTarget,
     });
     return;
-  } else {
-    // Apply the @list decorator to the operation
-    context.call(typespecDecorators.TypeSpec.list, target);
   }
+  
+  // Check the response model for @pageItems decorator
+  const responseModel = hasModelResponse.type as Model;
+  let hasPageItemsProperty = false;
+  
+  // Check if any property has @pageItems decorator
+  for (const [, prop] of responseModel.properties) {
+    // Check if the property is marked with @pageItems by checking the program state
+    // The @pageItems decorator uses a state symbol "pageItems"
+    const pageItemsStateKey = Symbol.for("TypeSpec.pageItems");
+    if (context.program.stateSet(pageItemsStateKey).has(prop)) {
+      hasPageItemsProperty = true;
+      break;
```

Comment:
to check if a property has this decorator, we should use this function from typespec/compiler: `isPageItemsProperty`